crates/jsonschema/src/lib.rs

```
@@ -1140,88 +1133,71 @@ pub mod meta {
-    ///
-    /// # Returns
-    ///
-    /// - `Ok(Ok(()))` if the schema is valid
-    /// - `Ok(Err(ValidationError))` if the schema is invalid
-    /// - `Err(ReferencingError)` if the meta-schema can't be detected
-    ///
-    /// # Examples
-    ///
-    /// ```rust
-    /// use serde_json::json;
-    ///
-    /// let schema = json!({
-    ///     "type": "string",
-    ///     "maxLength": 5
-    /// });
-    /// assert!(jsonschema::meta::try_validate(&schema).expect("Invalid schema").is_ok());
-    ///
-    /// // Invalid schema
-    /// let invalid_schema = json!({
-    ///     "type": "invalid_type"
-    /// });
-    /// assert!(jsonschema::meta::try_validate(&invalid_schema).expect("Invalid schema").is_err());
-    ///
-    /// // Invalid $schema URI
-    /// let undetectable_schema = json!({
-    ///     "$schema": "invalid-uri",
-    ///     "type": "string"
-    /// });
-    /// assert!(jsonschema::meta::try_validate(&undetectable_schema).is_err());
-    /// ```
-    ///
-    /// # Errors
-    ///
-    /// Returns an error when the draft cannot be detected (for example, because `$schema` contains an invalid URI).
-    pub fn try_validate(
-        schema: &Value,
-    ) -> Result<Result<(), ValidationError<'_>>, ReferencingError> {
-        Ok(try_meta_validator_for(schema)?.as_ref().validate(schema))
-    }
+    fn resolve_meta_schema_chain(
+        uri: &str,
+    ) -> Result<(Value, Draft), Box<dyn std::error::Error + Send + Sync>> {
+        use ahash::AHashSet;
+        use referencing::Retrieve;
+
+        let mut visited = AHashSet::new();
+        let mut current_uri = uri.to_string();
+        let retriever = crate::retriever::DefaultRetriever;
+        let first_meta_uri = referencing::uri::from_str(&current_uri)?;
+
+        // Attempt to retrieve the first meta-schema. If this fails (e.g., network error,
+        // unreachable URI), we'll fall back to Draft 2020-12 validation.
```

Maybe don't have such a duality? and fail loudly instead?